reaper/reaper/src/main/kotlin/com/emergetools/reaper/ReaperReportUploadWorker.kt

```diff
@@ -175,11 +177,14 @@ internal class ReaperReportUploadWorker(
       stream.write(reportString.encodeToByteArray())
     }
 
+    val compressedReport = gzipCompressReport(reportString)
+
     val url = "$baseUrl/report"
     val request = Request.Builder().apply {
       header("Authorization", "Bearer $apiKey")
+      header("Content-Encoding", "gzip")
       url(url)
-      post(reportString.toRequestBody("application/json; charset=utf-8".toMediaTypeOrNull()))
+      post(compressedReport.toRequestBody("application/json; charset=utf-8".toMediaTypeOrNull()))
```

Comment:
I think okhttp has a method for this: https://square.github.io/okhttp/5.x/okhttp/okhttp3/-request-body/-companion/gzip.html

Something like:
```
val body = reportString.toRequestBody("application/json; charset=utf-8".toMediaTypeOrNull());
val request = Request.Builder().apply {
  header("Authorization", "Bearer $apiKey")
  header("Content-Encoding", "gzip")
  url(url)
  post(body.gzip())
};
```

I think

Reply:
Appears to only be in the 5.0 snapshot, which we're on 4.11. I'll see if 4.x has one, as I'd prefer to not leverage a snapshot release.

Reply:
Follow up, did some reading and despite 5.x-snapshot state, it sounds like it actually is quite stable/no breaking changes. 

We need to land backend support for this, but I can follow up with updating OkHTTP to 5.x and leverage the helper function.